### PR TITLE
frontend/fix : #658 Extracted device details and added to headers

### PR DESCRIPTION
**Pull request by AmritpalSingh6749**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added device details to API headers

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by hkmangla** [comment]
Please follow contribution guidelines in commit message and PR title.

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 1]
```diff
@@ -263,6 +265,108 @@ public void onBackPressed() {
...
+            String deviceRAM = getDeviceRAM();
+
+            if(bModel == null || bModel == "")
+                bModel="null";
+            if(bBrand == null || bBrand == "")
+                bBrand="null";
+            if(bVersion == null || bVersion == "")
+                bVersion = "null";
+            else
+                bVersion = "Android v"+bVersion;
```
Comment (ashishsingh101) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:355]: use ternary operator here

Comment (ashishsingh101) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:355]: other places as well where you can optimize

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 2]
```diff
@@ -263,6 +265,108 @@ public void onBackPressed() {
...
+
+            deviceDetails+=bBrand+"/";
+            deviceDetails+=bModel+"/";
+            deviceDetails+=bVersion+"/";
+            deviceDetails+=deviceRAM + "/";
+            deviceDetails+=dim[1]+"/";
+            deviceDetails+=dim[0];
```
[the remark this answers is not in the document]
Comment (AmritpalSingh6749) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java]: done

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java (hunk 1 above)
Comment (AmritpalSingh6749) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java]: done

**Review by ashishsingh101 — APPROVED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**